exporter/exporterhelper/common.go

```diff
@@ -16,16 +16,65 @@ package exporterhelper
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	"go.opencensus.io/trace"
 
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenterror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer/consumererror"
 )
 
 var (
 	okStatus = trace.Status{Code: trace.StatusCodeOK}
 )
 
+type TimeoutSettings struct {
+	// Timeout is the timeout for each operation.
+	Timeout time.Duration `mapstructure:"timeout"`
+}
+
+func CreateDefaultTimeoutSettings() TimeoutSettings {
+	return TimeoutSettings{
+		Timeout: 5 * time.Second,
+	}
+}
+
+type settings struct {
+	configmodels.Exporter
+	TimeoutSettings
+	QueuedSettings
+	RetrySettings
+}
+
+type request interface {
```

Comment:
Please add a comment that explains the struct.

Reply:
Done.